ios/FluentUI.Demo/FluentUI.Demo/Demos/ColorDemoController.swift

```diff
@@ -139,36 +135,31 @@ class ColorDemoController: UIViewController {
     }
 
     private lazy var segmentedControl: SegmentedControl = {
-        let segmentedControl = SegmentedControl(items: colorProviderThemes.map({ return SegmentItem(title: $0.name) }), style: .primaryPill)
+        let segmentedControl = SegmentedControl(items: DemoColorTheme.allCases.map({ return SegmentItem(title: $0.name) }), style: .primaryPill)
         segmentedControl.addTarget(self, action: #selector(segmentedControlValueChanged(sender:)), for: .valueChanged)
         return segmentedControl
     }()
 
     @objc private func segmentedControlValueChanged(sender: Any) {
         if let segmentedControl = sender as? SegmentedControl {
-            let windowType = colorProviderThemes[segmentedControl.selectedSegmentIndex].demoColorTheme.windowType
-            let colorThemeHost = view.window?.windowScene?.delegate as? ColorThemeHosting
-            ColorDemoController.themeWindowType = windowType
-
-            if let navigationController = navigationController {
-                navigationController.popViewController(animated: false)
-                colorThemeHost?.updateToWindowWith(type: windowType, pushing: self)
+            let selectedSegmentIndex = segmentedControl.selectedSegmentIndex
+            let demoColorTheme = DemoColorTheme.allCases[selectedSegmentIndex]
+            let window = self.view.window
+
+            if let provider = demoColorTheme.provider {
+                let demoListViewController = navigationController?.viewControllers.first as? DemoListViewController
+                demoListViewController!.updateColorProviderFor(window: window!, provider: provider, themeName: demoColorTheme.name)
```

Comment:
Another force unwrap, similar to the one before. Make this assignment part of the previous `if let` condition, and you should be good to go.

Reply:
Actually I see two here. There's no guarantee that any view is currently hosted in a window, so `window!` is never safe. Since everything here depends on `window` existing, it might make sense to put all of this provider code in an `if let window = self.view.window` since none of it will work otherwise.